service/history/workflow/mutable_state_impl.go

```diff
@@ -2846,17 +2846,26 @@ func (ms *MutableStateImpl) UpdateBuildIdAssignment(buildId string) error {
 	// because build ID is changed, we clear sticky queue so to make sure the next wf task does not go to old version.
 	ms.ClearStickyTaskQueue()
 	limit := ms.config.SearchAttributesSizeOfValueLimit(ms.namespaceEntry.Name().String())
-	return ms.updateBuildIdsSearchAttribute(&commonpb.WorkerVersionStamp{UseVersioning: true, BuildId: buildId}, limit)
+	return ms.updateBuildIdsAndDeploymentSearchAttributes(&commonpb.WorkerVersionStamp{UseVersioning: true, BuildId: buildId}, limit)
 }
 
-// For pinned workflows (ms.GetEffectiveVersioningBehavior() == PINNED), this will append a tag formed as
-// pinned:<deployment_series_name>:<deployment_build_id> to the BuildIds search attribute,
-// if it does not already exist there. The deployment will be execution_info.deployment, or the override deployment if
-// a pinned override is set.
-// For all other workflows (ms.GetEffectiveVersioningBehavior() != PINNED), this will append a tag based on the
-// workflow's versioning status.
-func (ms *MutableStateImpl) updateBuildIdsSearchAttribute(stamp *commonpb.WorkerVersionStamp, maxSearchAttributeValueSize int) error {
-	changed, err := ms.addBuildIdToSearchAttributesWithNoVisibilityTask(stamp, maxSearchAttributeValueSize)
+// Sets TemporalWorkerDeployment to the override DeploymentName or`ms.executionInfo.WorkerDeploymentName` if present.
+// Sets TemporalWorkerDeploymentVersion to the override PinnedVersion or `ms.executionInfo.VersioningInfo.Version` if present.
+// Sets TemporalWorkflowVersioningBehavior to the override Behavior or `ms.executionInfo.VersioningInfo.Behavior` if specified.
```

Comment:
this should be Pinned if override PinnedVersion is present

Reply:
oh yes. `GetEffectiveBehavior` is what is implemented, the comment is just wrong. will fix

Reply:
the comment says `the override Behavior or `ms.executionInfo.VersioningInfo.Behavior` if specified`, do you think it's not clear that override takes priority?

Reply:
it's good

Reply:
I fixed the sentence